File: rgctl-tests/ecommerce-csharp/src/Ecommerce/Coolstore/Models/CatalogProduct.cs
namespace Ecommerce.Coolstore.Models;

/** Lightweight CoolStore catalog product (itemId keyed). */
public class CatalogProduct
{
    public string ItemId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Desc { get; set; } = string.Empty;
    public double Price { get; set; }

    public CatalogProduct()
    {
    }

    public CatalogProduct(string itemId, string name, string desc, double price)
    {
        ItemId = itemId;
        Name = name;
        Desc = desc;
        Price = price;
    }
}

File: rgctl-tests/ecommerce-java/src/main/java/com/example/ecommerce/coolstore/model/CatalogProduct.java
package com.example.ecommerce.coolstore.model;

/** Lightweight CoolStore catalog product (itemId keyed). */
public class CatalogProduct {
    private String itemId;
    private String name;
    private String desc;
    private double price;

    public CatalogProduct() {}

    public CatalogProduct(String itemId, String name, String desc, double price) {
        this.itemId = itemId;
        this.name = name;
        this.desc = desc;
        this.price = price;
    }

    public String getItemId() { return itemId; }
    public void setItemId(String itemId) { this.itemId = itemId; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getDesc() { return desc; }
    public void setDesc(String desc) { this.desc = desc; }
    public double getPrice() { return price; }
    public void setPrice(double price) { this.price = price; }
}
